flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/AccessExecutionGraph.java

```diff
@@ -199,4 +206,12 @@ public interface AccessExecutionGraph extends JobStatusProvider {
      * @return The changelog storage name, or an empty Optional in the case of batch jobs
      */
     Optional<String> getChangelogStorageName();
+
+    /**
+     * Retrieves the count of pending operators waiting to be transferred to job vertices in the
+     * adaptive execution of batch jobs.
+     *
+     * @return the number of pending operators.
+     */
+    int getPendingOperatorCount();
```

Comment:
if this is batch specific - batch should be in the method name

Reply:
I think the method name should be general ones in `AccessExecutionGraph`. Since it defines the interface to retrieve the runtime informations of the execution graph and doesn't distinguish whether it's a streaming or a batch job.